writer.go

```diff
@@ -84,6 +91,23 @@ func open(paths []string) ([]zapcore.WriteSyncer, func(), error) {
 	return writers, closeAll, nil
 }
 
+func checkOpenPaths(paths []string) error {
+	var openErr error
+	for _, path := range paths {
+		if !strings.HasPrefix(path, "file:") {
+			continue
+		}
+		u, err := url.Parse(path)
+		if err != nil {
+			return err
+		}
+		if !(filepath.IsAbs(u.Path) && !strings.Contains(u.Path, "..")) {
```

Comment:
if filepath.IsAbs returns true, we're good. It's okay if an absolute path has `/../` because that'll just modify the absolute path. `/foo/bar/../baz` is just `/foo/baz`.

Reply:
@abhinav, I think that means we don't need this feature then?

`sr.openFile`, given a rawUrl with a file scheme, always consumes an absolute path because `url.Parse` return object's Path field only contains an absolute path under this constraints. `filepath.IsAbs` will always return true here.

Experimentation:
https://go.dev/play/p/Xcjj0PhmgXo

If that make sense I can close the issue?

Reply:
@r-hang can we confirm with a test that it's not possible to provide a relative path or one with '..' to that API?

Reply:
https://github.com/uber-go/zap/pull/1398 as a successor since I'm not able to reopen this one.